AlphaWallet/WhatsNew/ViewControllers/WhatsNewListingViewController.swift

```diff
@@ -46,7 +46,9 @@ extension WhatsNewListingViewController: ModalViewControllerDelegate {
 
     func didClose(_ controller: ModalViewController) {
         controller.dismissViewAnimated {
```

Comment:
The logic seems a bit strange. From reading at this client code, it appears that `controller` has to be dismissed twice. Is this something that should be fixed in ModalViewController instead?

Reply:
I'm wary of fixing it there because I don't know what code which uses it is doing. I might have to go fix all the instances  where it's used. I'll post another issue and make it an app wide change if you want.

Reply:
Ah, good point. See if it's a problem elsewhere then. 

> > I think, it's better to include a screenshot which illustrates the problem, a bit easier/faster to grasp it even before running the reproduction steps.
> 
> I put two images in #3520 that sorts of explains the problem. Next time I will put in a movie.

Great! But if images display the issue, use that instead of movies. The latter is slower to comprehend.

Reply:
Four instances use this. 
1. WalletConnectConnectionTimeoutViewController
2. SwitchChainRequestViewController
3. PromptViewController
4. CollectUsersEmailViewController
5. WhatsNewListingViewController

All four execute a variation of the following:
```
        dismissViewAnimated(with: {
            self._delegate?.didClose(in: self)
            self.dismiss(animated: false)
        })
```

Reply:
Ah, I see. Maybe create a separate issue/PR to move the call for `elf.dismiss(animated:)` into `dismissViewAnimated()` and modify the name if necessary?

But for this PR, just fix this particular case. But can you double check it this: the 4 examples you listed do not access `controller`, they refer to `self` instead.

Reply:
@hboon Here `self` is the same as `controller` since the other classes are delegates for themselves and this delegate is set in the `ModalViewController`'s `init` function. If you did change the delegate in the other classes, the code won't work.